controllers/userController.js

```diff
@@ -256,3 +256,17 @@ export const updateUserRole = async (req, res, next) => {
     next(new ErrorHandler("Error updating user role", 400));
   }
 };
+
+export const addCourse = catchAsyncError (async(req, res) => {
+  // Extract Course Id
+  const courseId = req.params.id;
+  const course = await Course.find({_id : courseId});
+
+  const userId = req.user?._id;
+  const user = await User.find({_id : userId});
+
```

Comment:
Handle the Error if course is not found.

Reply:
I have also used a middleware to check whether user is logged in or not that can easily handle that error

Reply:
> I have also used a middleware to check whether user is logged in or not that can easily handle that error

Nor for auth, I am telling if the course is not present. It will become developer freindly and easy to debug for errors

Reply:
> Handle the Error if course is not found.

Handle for
1 if course is not present in db
2 id course is already in the playlist